samza-core/src/main/java/org/apache/samza/clustermanager/ClusterBasedJobCoordinator.java

```diff
@@ -242,9 +249,8 @@ private JobModelManager buildJobModelManager(Config coordinatorSystemConfig, Met
     return jobModelManager;
   }
 
-  private StreamPartitionCountMonitor getPartitionCountMonitor(Config config) {
-    Map<String, SystemAdmin> systemAdmins = new JavaSystemConfig(config).getSystemAdmins();
-    StreamMetadataCache streamMetadata = new StreamMetadataCache(Util.javaMapAsScalaMap(systemAdmins), 0, SystemClock.instance());
+  private StreamPartitionCountMonitor getPartitionCountMonitor(Config config, Map<String, SystemAdmin> systemAdminMap) {
+    StreamMetadataCache streamMetadata = new StreamMetadataCache(Util.javaMapAsScalaMap(systemAdminMap), 0, SystemClock.instance());
```

Comment:
Should this take SystemAdmins instead of systemAdminMap?

Reply:
I think it is better to use SystemAdmins in StreamMetadataCache. Since it is mostly cosmetic change to the constructor of `StreamMetadataCache`, I have not done this in the current patch to make the review easier. Do you prefer me to make the change here? I can do it.

Reply:
Fixed as suggested.